## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 438533`

Logs: https://github.com/daspk04/nixpkgs-review-gha/actions/runs/17342125896


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.syne-tune</li>
    <li>python312Packages.syne-tune.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.syne-tune</li>
    <li>python313Packages.syne-tune.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.syne-tune</summary>
<pre>tst/schedulers/test_botorch_transfer.py::test_transfer_samples_added_and_encoding[True-task_id_list2]
tst/schedulers/test_botorch_transfer.py::test_transfer_samples_added_and_encoding[True-task_id_list2]
  /build/source/tst/schedulers/test_botorch_transfer.py:44: DeprecationWarning: Bitwise inversion '~' on bool is deprecated and will be removed in Python 3.16. This returns the bitwise inversion of the underlying int object and is usually not what you expect from negating a bool. Use the 'not' operator for boolean negation or ~int(x) if you really want the bitwise inversion of the underlying int.
    return ((encoding >= 0) & (encoding <= 1) & ~pd.isna(encoding)).all()

tst/schedulers/test_botorch_transfer.py::test_transfer_samples_added_and_encoding[False-task_id_list0]
tst/schedulers/test_botorch_transfer.py::test_transfer_samples_added_and_encoding[False-task_id_list1]
tst/schedulers/test_botorch_transfer.py::test_transfer_samples_added_and_encoding[True-task_id_list2]
  /build/source/tst/schedulers/test_botorch_transfer.py:95: DeprecationWarning: __array__ implementation doesn't accept a copy keyword, so passing copy=False failed. __array__ must implement 'dtype' and 'copy' keyword arguments. To learn more, see the migration guide https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword
    for encoding in np.array(scheduler.searcher._config_to_feature_matrix(old_confs)):

tst/test_config_space.py::test_convert_config_space
tst/test_config_space.py::test_convert_config_space
  /nix/store/f84zsc3qnqscriz65by22lj95s11z0g1-python3.12-ray-2.49.0/lib/python3.12/site-packages/ray/tune/search/sample.py:178: RayDeprecationWarning: The `base` argument is deprecated. Please remove it as it is not actually needed in this method.
    _warn_for_base()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tst/experiments/test_plotting.py::test_filter_final_row_per_trial - Failed: Timeout (>1.0s) from pytest-timeout.
= 1 failed, 553 passed, 87 skipped, 4 deselected, 21 warnings in 104.05s (0:01:44) =</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.syne-tune</li>
    <li>python312Packages.syne-tune.dist</li>
    <li>python313Packages.syne-tune</li>
    <li>python313Packages.syne-tune.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.syne-tune</li>
    <li>python312Packages.syne-tune.dist</li>
    <li>python313Packages.syne-tune</li>
    <li>python313Packages.syne-tune.dist</li>
  </ul>
</details>

--- a/pkgs/development/python-modules/syne-tune/default.nix
+++ b/pkgs/development/python-modules/syne-tune/default.nix
@@ -1,0 +1,146 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  setuptools,
+
+  # dependencies
+  dill,
+  numpy,
+  pandas,
+  sortedcontainers,
+  typing-extensions,
+
+  # optional-dependencies
+  autograd,
+  botorch,
+  # configspace,
+  fastparquet,
+  h5py,
+  huggingface-hub,
+  matplotlib,
+  onnxruntime,
+  pymoo,
+  pyyaml,
+  scikit-learn,
+  scipy,
+  # smac,
+  statsmodels,
+  swig,
+  xgboost,
+  # yahpo-gym,
+
+  # tests
+  pytestCheckHook,
+  pytest-timeout,
+  ray,
+  writableTmpDirAsHomeHook,
+}:
+buildPythonPackage rec {
+  pname = "syne-tune";
+  version = "0.14.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "syne-tune";
+    repo = "syne-tune";
+    tag = "v${version}";
+    hash = "sha256-51QyfJ0XOcXTeE95YOhtUmhat23joaEYvUnk7hYfksY=";
+  };
+
+  postPatch = ''
+    substituteInPlace syne_tune/optimizer/schedulers/synchronous/hyperband.py \
+     --replace-fail 'metric_val=np.NAN' 'metric_val=np.nan'
+    substituteInPlace syne_tune/optimizer/schedulers/synchronous/dehb.py \
+     --replace-fail 'result_failed.metric_val = np.NAN' 'result_failed.metric_val = np.nan'
+  '';
+
+  build-system = [
+    setuptools
+  ];
+
+  pythonRelaxDeps = [
+    "numpy"
+  ];
+
+  dependencies = [
+    dill
+    numpy
+    pandas
+    sortedcontainers
+    typing-extensions
+  ];
+
+  optional-dependencies = lib.fix (self: {
+    blackbox-repository = [
+      fastparquet
+      h5py
+      huggingface-hub
+      numpy
+      pandas
+      scikit-learn
+      xgboost
+    ];
+    bore = [
+      scikit-learn
+      xgboost
+    ];
+    botorch = [ botorch ];
+    gpsearchers = [
+      autograd
+      scipy
+    ];
+    kde = [ statsmodels ];
+    moo = [
+      pymoo
+      scipy
+    ];
+    sklearn = [ scikit-learn ];
+    # smac = [ smac swig ]; # smac unavailable on nixpkgs
+    visual = [ matplotlib ];
+    # yahpo = [ configspace onnxruntime pandas pyyaml yahpo-gym ]; # yahpo-gym unavailable on nixpkgs
+  });
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-timeout
+    ray
+    writableTmpDirAsHomeHook
+  ]
+  ++ ray.optional-dependencies.tune
+  ++ optional-dependencies.blackbox-repository
+  ++ optional-dependencies.bore
+  ++ optional-dependencies.botorch
+  ++ optional-dependencies.gpsearchers
+  ++ optional-dependencies.kde
+  ++ optional-dependencies.sklearn;
+
+  disabledTests = [
+    # NameError: name 'HV' is not defined
+    # these require pkg `pymoo` and adding `pymoo` raises:
+    # setuptools.errors.PackageDiscoveryError: Multiple top-level packages discovered in a flat-layout: ['cma', 'notebooks'].
+    "test_append_hypervolume_indicator"
+    "test_hypervolume"
+    "test_hypervolume_progress"
+    "test_hypervolume_simple"
+  ];
+
+  disabledTestPaths = [
+    # legacy test
+    "tst/schedulers/test_legacy_schedulers_api.py"
+  ];
+
+  pythonImportsCheck = [
+    "syne_tune"
+  ];
+
+  meta = {
+    description = "Large scale asynchronous hyperparameter and architecture optimization library";
+    homepage = "https://github.com/syne-tune/syne-tune";
+    changelog = "https://github.com/syne-tune/syne-tune/releases/tag/${src.tag}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ daspk04 ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -17740,6 +17740,8 @@ self: super: with self; {
     }
   );
 
+  syne-tune = callPackage ../development/python-modules/syne-tune { };
+
   synergy = callPackage ../development/python-modules/synergy { };
 
   synology-srm = callPackage ../development/python-modules/synology-srm { };

